pam/go-loader/meson.build

```diff
@@ -0,0 +1,67 @@
+project('pam-go-loader', 'c',
```

Comment:
I know you mention meson in your PR explicitely. I suggest that we try to get without it for now and only rely on standard go tooling to build the project.

What we did in the past for building other modules is generally abusing `go generate` with a dest directory. If set, it means build and copy in that destdir, otherwise do nothing.

That way, we won’t request anyone working on this repo to install anything than Go + the headers/gcc.

Reply:
Well, I see this and I'm also abusing of `go generate` for the module generation, but at the same time I feel that we then have something that is less maintainable.

IMHO we should indeed have `go generate` around so that it's possible to just hack on things with pure go, but at the same time have a proper build / install system that generates stuff at install time and that it's reproducible. Leaving the job of `debian/rules` to be just a single line. Sure this can be done all in `go` too, but looks like a bit an over-complication.

In this case I've just used meson because it allowed me to have a simple way to generate files (and we do need this) and configure binaries.

What I had ideally done, but I skipped because we need more discussion would have been:
 - Add a meson project that generates the authd pam library
 - generate the pam-loader library passing (even hardcoded via `-D`) the final loaded library path
 - Install them both were required

And this can be easily extended to the other parts of the project too, as said... The main goal here is making this installable in a reproducible an maintainable way.

However, this can be re-discussed deeply, for this specific case we can use a simple generator probably but this one was the faster way.

Reply:
Right, I’m not too found of introducing another tool that most of the team doesn’t know (especially when it can be a one liner), but have an easy "run that tool in that mode", which will install the assets too (even if I agree, this is abusing it).

I clearly understand that meson is giving us more oppurtunities of integration, and was quicker I guess for you in that case, using pkg-config and such (this is why I implemented the Yaru theme and sync from gtk and gnome shell theming in meson), however, I would still resist to it a little bit until we have a clear need for it, sounds ok?

Reply:
Sure, fair enough... I won't push on this, we can revisit if needed.